### PR TITLE
[CI, tasks] Use `go install` to install tooling dependencies

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Use `go install` instead of `go run` to install dependencies.

### Motivation

- `go run` doesn't interact well  with nested Go modules (you need to pin tooling version on every module)
- Tooling dependencies should not be added to the licenses list, separating them fixes this.

### Additional Notes

- A new task `install-tools` needs to be run before using any of the Go tooling.
- I kept `go:generate` tooling on the main module, maybe we can move it? I was hesitant because we don't want, say, the tooling`easyjson`dependency we use on `go:generate` to get out of sync with the one that we import in the code.

### Describe your test plan

Pipeline passes

### CODE REVIEW

File: .gitlab/package_build/apk.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ agent_android_apk:
     - inv -e deps-vendored
     - cd $SRC_PATH
     - python3 -m pip install -r requirements.txt
+    - inv -e install-tools
```
Comment (mx-psi) [inline .gitlab/package_build/apk.yml:24]:
```suggestion
    - inv -e install-tools
    - inv -e deps-vendored
    - cd $SRC_PATH
    - python3 -m pip install -r requirements.txt
```

**Review by KSerrania — APPROVED** [review]

**Review by KSerrania — COMMENTED** [review]
If the `inv -e install-tools` step is mandatory to run go tooling, this should be added to the developer documentation.

**Review by ahmed-mez — APPROVED** [review]
Changes in `pkg/clusteragent/README.md` LGTM

**Review by juliogreff — APPROVED** [review]

**Comment by albertvaka** [comment]
Do we need a new `inv install-tools` task, or could we do this as part of `inv deps`?

**Comment by KSerrania** [comment]
> Do we need a new inv install-tools task, or could we do this as part of inv deps?

We can put everything in `inv deps`. From the way the PR is structured, I believe the reason to put the tool install in a separate task is that the tools installed by `inv install-tools` are only needed for testing (you can see that no build job in the CI installs the tools); whereas `inv deps` downloads dependencies for building the Agent.

I don't have a strong opinion either way. I like the fact that we're dissociating test and build dependencies that way, but it does require one additional step for users. We could also:
- change the task name to specify that these tools are only needed for testing purposes (eg. `inv install-testing-tools`)
- add an option to `inv deps` to also install build tools, instead of making it a separate task.

**Review by olivielpeau — APPROVED** [review]
Only reviewed owned file (`cmd/agent/android/README.md`), lgtm, feel free to add the docs team as owners of that file

**Review by albertvaka — APPROVED** [review]

File: pkg/clusteragent/README.md [hunk 2]
```diff
@@ -27,11 +27,12 @@ For pre-requisite, refer to the Agent's Getting Started section in the [README](
 
 To start working on the Cluster Agent, you can build the `master` branch:
 
-1. checkout the repo: `git clone https://github.com/DataDog/datadog-agent.git $GOPATH/src/github.com/DataDog/datadog-agent`.
+1. Checkout the repo: `git clone https://github.com/DataDog/datadog-agent.git $GOPATH/src/github.com/DataDog/datadog-agent`.
```
Comment (jtappa) [inline pkg/clusteragent/README.md:30]:
```suggestion
1. Clone the repo: `git clone https://github.com/DataDog/datadog-agent.git $GOPATH/src/github.com/DataDog/datadog-agent`.
```

**Pull request merged**